### PR TITLE
Remove our dependency on Microsoft.Toolkit.Win32.UI.XamlApplication

### PR DESCRIPTION
**Pull request by DHowett**
We originally needed this library (or a separate DLL in our own project)
to handle hooking up the XAML resource loader to the providers that our
application needed. It was introduced in its nascent form in 2019, in a
PR titled "Make XAML files work."

It appears we no longer need it, and the provider hookup is being
handled by our `AppT2` base class override. I've tested this in Windows
10 Vb running unpackaged, and it seems to work totally fine. Crazy.

Removing this dependency saves us a couple hundred kilobytes on disk and
removes one consumer of the App CRT from our package.

### CODE REVIEW

File: scratch/ScratchIslandApp/SampleApp/App.cpp [hunk 1]
```diff
@@ -33,6 +26,44 @@ namespace winrt::SampleApp::implementation
...
+        {
+            _windowsXamlManager = xaml::Hosting::WindowsXamlManager::InitializeForCurrentThread();
+        }
+        else
+        {
+            _isUwp = true;
+        }
+    }
+
+    void App::Close()
```
Comment (DHowett) [inline scratch/ScratchIslandApp/SampleApp/App.cpp:42]: behavior that was originally trapped in the toolkit application is herein merged (same for TerminalApp)

File: scratch/ScratchIslandApp/SampleApp/App.cpp [hunk 2]
```diff
@@ -12,19 +12,12 @@ using namespace winrt::Windows::UI::Xaml;
 using namespace winrt::Windows::UI::Xaml::Controls;
 using namespace winrt::Windows::UI::Xaml::Navigation;
 
+namespace xaml = ::winrt::Windows::UI::Xaml;
+
 namespace winrt::SampleApp::implementation
 {
     App::App()
     {
-        // This is the same trick that Initialize() is about to use to figure out whether we're coming
```
Comment (zadjii-msft) [inline scratch/ScratchIslandApp/SampleApp/App.cpp:19]: this comment is still relevant

**Review by zadjii-msft — APPROVED** [review]

**Comment by awakecoding** [comment]
🚀 thanks a lot for these changes, this will bring us closer to properly supporting a Windows Terminal build option without the dynamic visual studio runtime! 🙏🪄

**Review by carlos-zamora — APPROVED** [review]

**Comment by carlos-zamora** [comment]
@msftbot merge this in 10 minutes

**Comment by DHowett** [comment]
I trust this change about as far as I could throw it... so I'd love to get a good validation pass on the release build after it lands :)

**Comment by sylveon** [comment]
FWIW, I have been running a XAML Islands app in prod without that dependency for years :)

I've never liked that dependency so quickly worked getting rid of it once I realized it never did much.

https://github.com/TranslucentTB/TranslucentTB/commit/021ede702290573d871f742125ed5835e8a76de9

**Comment by DHowett** [comment]
@sylveon thanks so much for that. Gives me the confidence to backport this to stable and remove one more variable from our troubleshooting pipeline :)

**Comment by zadjii-msft** [comment]
Could removing this have caused #14729? I'd reckon it's possible. Maybe there's an interplay between this and the FWP/Win11 version of the package.

**Pull request merged**